### PR TITLE
Simplify testKitGen

### PR DESCRIPTION
**Pull request by renfeiw**
- utilize make recursive structure
- separate settings.mk from makefile
- separate tag into level and group
- add disabled tag
- remove openJDK test
- support makefile include

Signed-off-by: Renfei Wang <renfeiw@ca.ibm.com>

### CODE REVIEW

**Comment by renfeiw** [comment]
reviewers: @smlambert @llxia

**Comment by llxia** [comment]
Jenkins test sanity

File: test/README.md [hunk 1]
```diff
@@ -109,27 +109,17 @@ make test
 
   * sanity tests
     ```
-    make sanity
-    ```
-
-  * extended tests
-    ```
-    make extended
+    make _sanity
```
Comment (smlambert) [inline test/README.md]: is there a reason not to continue to support the make targets without the '_' prefix?

File: test/README.md [hunk 2]
```diff
@@ -186,6 +177,18 @@ make test
 
 5. Exclude tests:
 
+  * exclude test target in playlist.xml
+
+    Add 
+    ```
+    <disabled>reason for disabled<disabled>
```
Comment (smlambert) [inline test/README.md]: <disabled>Reason for disabling test, preferably includes issue number<**/**disabled>

May additionally want to include example

File: test/TestConfig/settings.mk [hunk 3]
```diff
@@ -0,0 +1,247 @@
...
+# Define the EXCLUDE_FILE to be used for temporarily excluding failed tests.
+# This macro is used in /test/Utils/src/org/openj9/test/util/IncludeExcludeTestAnnotationTransformer
+ifndef EXCLUDE_FILE
+	export EXCLUDE_FILE:=$(JVM_TEST_ROOT)$(D)TestConfig$(D)resources$(D)excludes$(D)$(JCL_VERSION)_exclude_$(JAVA_VERSION).txt
+endif
+
+#######################################
+# failed target
+#######################################
+FAILEDTAGETS = $(TESTOUTPUT)$(D)failedtargets.mk
```
Comment (smlambert) [inline test/TestConfig/settings.mk]: would love to see this spelling mistake fixed, should be FAILEDTARGETS with an R

File: test/TestConfig/settings.mk [hunk 4]
```diff
@@ -0,0 +1,247 @@
...
+#  SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+##############################################################################
+
+.PHONY: help rmResultFile resultsSummary
+
+help:
+	@echo "This makefile is used to build and execute JVM tests. You should specify the following"
+	@echo "variables before using this script:"
+	@echo "JAVA_BIN  Path to java bin dir which will be used to built and executed JVM tests"
+	@echo "SPEC      Should match the current platform (e.g. SPEC=linux_x86-64, SPEC=linux_x86, SPEC=linux_x86-64)"
```
Comment (smlambert) [inline test/TestConfig/settings.mk]: you have a duplicate entry in your e.g. list SPEC=linux_x86-64

File: test/README.md (hunk 1 above)
Comment (renfeiw) [reply test/README.md]: We still support target without '_', but would not recommend the use of it as it doesn't print java version and result summary.

**Comment by renfeiw** [comment]
@smlambert updated based on your comment

**Comment by renfeiw** [comment]
Jenkins test sanity

File: test/README.md (hunk 1 above)
Comment (smlambert) [reply test/README.md]: We should reverse that at some point (so the non _ version gives you the extras).

**Comment by smlambert** [comment]
I think this additionally changes the dependencies list, if so, not sure I saw XML parser module added to the README.

**Comment by llxia** [comment]
Jenkins test sanity

**Comment by renfeiw** [comment]
The XML parser is not changed in this update, cause the machines is still not configured to the right point. We will make sure to modify the README in the following PR with XML parser change.

File: test/README.md (hunk 1 above)
Comment (renfeiw) [reply test/README.md]: Thanks. It's not a straight forward fix as we take advantage of make to add extra targets. We will think about this in the next update. We will modify the Jenkins configuration along with this PR.

File: test/Java8andUp/playlist.xml [hunk 5]
```diff
@@ -717,7 +719,9 @@
 			<variation>NoOptions</variation>
 			<variation>-XX:RecreateClassfileOnload</variation>
 		</variations>
-		<command>$(JAVA_COMMAND) $(JVM_OPTIONS) -verbose:stacktrace -Djava.security.policy=$(Q)$(TEST_RESROOT)$(D)java.policy$(Q) \
+		<command>$(MKDIR) -p $(REPORTDIR); \
```
[the remark this answers is not in the document]
Comment (renfeiw) [reply test/Java8andUp/playlist.xml]: Thanks, will fix it in the following PR.

**Comment by llxia** [comment]
Jenkins test sanity

**Pull request merged**